packages/amplify_datastore_plugin_interface/lib/amplify_datastore_plugin_interface.dart

```diff
@@ -43,26 +43,58 @@ export 'src/types/models/subscription_event.dart';
 export 'src/publicTypes.dart';
 
 abstract class DataStorePluginInterface extends AmplifyPluginInterface {
-  late ModelProviderInterface modelProvider;
+  /// modelProvider
+  late final ModelProviderInterface modelProvider;
 
-  /// Constructs an AmplifyPlatform
-  DataStorePluginInterface({required Object token, required this.modelProvider})
+  /// Datastore sync interval (in seconds)
+  late final int? syncInterval;
+
+  /// Datastore max number of records to sync
+  late final int? syncMaxRecords;
+
+  /// Datastore page size to sync
+  late final int? syncPageSize;
```

Comment:
Should these fields actually be nullable for the use case of `tokenOnly`? as they will never be assigned within this use case

Reply:
+1 I think the `late` implies that these could be initialized post-constructor but we do not actually have mechanisms for that so we could just be creating situations where we assume something is non-nullable when it's actually very possible it's never defined (hence null).

Reply:
Shouldn't they be nullable because they won't always be passed in the constructor?

@HuiSF 

Agree with cshfang, I'll remove the late keyword

Reply:
Yes @fjnoyp I meant to remove `late` sorry for the confusion

Reply:
Great removed the late modifier